File: src/react-staff-admin/src/features/notification/domain/usecases/create_notification.usecase.ts
import { NotificationRepository } from '../repositories/notification.repository';
import { Notification } from '../entities/notification.entity';

export class CreateNotificationUseCase {
  constructor(private repository: NotificationRepository) {}

  async execute(payload: {
    userId?: string;
    title: string;
    body: string;
    type: Notification['type'];
    metadata?: Record<string, any>;
  }): Promise<Notification> {
    return this.repository.createNotification(payload);
  }
}

File: src/sports_management/modules/notification_module/lib/domain/usecases/create_notification_usecase.dart
import 'package:server_module/server_module.dart';
import '../repositories/notification_repository.dart';

class CreateNotificationUseCase {
  const CreateNotificationUseCase(this._repository);

  final AppNotificationRepository _repository;

  Future<BaseResponse<dynamic>> call({
    required String userId,
    required String title,
    required String body,
    String type = 'SYSTEM',
  }) {
    return _repository.createNotification({
      'userId': userId,
      'title': title,
      'body': body,
      'type': type,
    });
  }
}
